kythe/go/platform/tools/kzip/kzip.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Binary kzip provides tools to work with kzip archives.
+//
+// Examples:
+//   # Merge 5 kzip archives into a single file.
+//   kzip merge --output output.kzip in{0,1,2,3,4}.kzip
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"kythe.io/kythe/go/platform/kzip"
+	"kythe.io/kythe/go/platform/vfs"
+	"kythe.io/kythe/go/util/flagutil"
+
+	"bitbucket.org/creachadair/stringset"
+)
+
+func init() {
+	flag.Usage = flagutil.SimpleUsage("Utility to modify/create kzip archives",
+		"merge --output path kzip-file*")
+	// TODO(schroederc): other subcommands
+}
+
+var output = flag.String("output", "", "Path to output kzip file")
+
+func main() {
+	flag.Parse()
+
+	if flag.Arg(0) != "merge" {
+		log.Printf("ERROR: unknown subcommand: %q", flag.Arg(0))
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	if err := flag.CommandLine.Parse(flag.Args()[1:]); err != nil {
+		log.Fatal(err)
+	}
+	if *output == "" {
+		log.Println("ERROR: required --output path missing")
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	if err := mergeArchives(context.Background(), *output, flag.Args()); err != nil {
+		log.Fatalf("Error merging archives: %v", err)
+	}
+}
+
+func mergeArchives(ctx context.Context, output string, archives []string) error {
+	out, err := vfs.Create(ctx, output)
+	if err != nil {
+		return fmt.Errorf("error creating output: %v", err)
+	}
+	wr, err := kzip.NewWriter(out)
```

Comment:
You could use https://godoc.org/kythe.io/kythe/go/platform/kzip#NewWriteCloser and combine the closes.